homeassistant/components/zone/__init__.py

```diff
@@ -40,29 +45,47 @@
 ICON_HOME = "mdi:home"
 ICON_IMPORT = "mdi:import"
 
-# The config that zone accepts is the same as if it has platforms.
-PLATFORM_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Required(CONF_LATITUDE): cv.latitude,
-        vol.Required(CONF_LONGITUDE): cv.longitude,
-        vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS): vol.Coerce(float),
-        vol.Optional(CONF_PASSIVE, default=DEFAULT_PASSIVE): cv.boolean,
-        vol.Optional(CONF_ICON): cv.icon,
-    },
+CREATE_FIELDS = {
+    vol.Required(CONF_NAME): cv.string,
```

Comment:
Should we accept empty strings?

Reply:
We had that for YAML but I actually made it a breaking change to require it. We should require names, how else do you know what the zone is about?

Reply:
I phrased it poorly.
What I meant: It is a required parameter, so should we require the name to be at least one character long? otherwise users can just use an empty string as the zone name, could they?

Reply:
although, in my test, the following configuration worked:
```
zone:
    - name: ""
      latitude: 32.8773367
      longitude: -117.2494053
      radius: 250
      icon: mdi:school
    - name: ""
      latitude: 2
      longitude: 2
      radius: 250
      icon: mdi:school
```
So I don't think this is an issue.